src/Doctrine/Orm/Extension/ParameterExtension.php

```diff
@@ -49,9 +54,20 @@ private function applyFilter(QueryBuilder $queryBuilder, QueryNameGeneratorInter
                 continue;
             }
 
-            $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            if (\is_string($filterId) && $this->filterLocator->has($filterId)) {
+                $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            }
+
+            if (\is_object($filterId)) {
```

Comment:
elseif?

Reply:
Here I've tried not to use the “else” keyword, according to Object Calisthenics "Don't Use The ELSE Keyword" principle. But I'm going to refactor it to look better.